### PR TITLE
finished task

### PR DESCRIPTION
**Pull request by nathanhuynhers**
[//]: # "Feel free to customize this template to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Added divs to separate the code into sections
- Added custom styling to these divs
- Added post image and centered it

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

image
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Review page.tsx and styles.modules.css

CC: @adityapawar1

### CODE REVIEW

File: app/styles.module.css [hunk 1]
```diff
@@ -10,7 +10,76 @@
...
+}
+
+.userDateHeader {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  margin: 10px;
+}
+
+.likesShareFooter {
```
Comment (adityapawar1) [inline app/styles.module.css:29]: Small nit: you can make this a little more general, like `postFooter`

File: app/styles.module.css [hunk 2]
```diff
@@ -10,7 +10,76 @@
...
+  max-width: 360px;
   height: 100%;
   border: 1px solid black;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  align-items: space-around;
+}
+
+.userDateHeader {
```
Comment (adityapawar1) [inline app/styles.module.css:22]: Same here: you can rename this to `postHeader`.

File: app/styles.module.css [hunk 3]
```diff
@@ -10,7 +10,76 @@
...
+}
+
+.separator {
+  border: none;
+  border-top: 1px solid #ddd;
+  margin: 10px 0;
+}
+
+.username {
+  margin-right: 110px;
```
Comment (adityapawar1) [inline app/styles.module.css:45]: Instead of putting a margin right here, you should use a space-between within a flex. You can put the profile placeholder and username in a div to group them. Then you can space between with that group and the date, which will move both objects to either side.

File: app/styles.module.css [hunk 4]
```diff
@@ -10,7 +10,76 @@
...
+  font-weight: bold;
+}
+
+.likes {
+  margin-right: 200px;
```
Comment (adityapawar1) [inline app/styles.module.css:50]: Same here, use a space-between instead of a large margin, since this can change based on screen size.

File: app/styles.module.css [hunk 5]
```diff
@@ -10,7 +10,76 @@
...
+}
+
+.date {
+  font-weight: 200;
+}
+
+.postImage {
+  width: 320px;
+  height: auto;
+  border-radius: 5%;
```
Comment (adityapawar1) [inline app/styles.module.css:62]: This should be set to 10px. You should generally be copying the styles from figma here, since we want everything to look as close to the design as possible.

File: app/styles.module.css [hunk 6]
```diff
@@ -10,7 +10,76 @@
...
+.likesShareFooter {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+  margin-right: 8px;
+  margin-left: 8px;
+  margin-bottom: 10px;
+}
+
+.separator {
```
Comment (adityapawar1) [inline app/styles.module.css:38]: Really like this solution for the separator!

**Review by adityapawar1 — CHANGES_REQUESTED** [review]
Couple of small style changes, overall looks amazing!

File: .env [hunk 7]
```diff
@@ -0,0 +1,2 @@
+NEXT_PUBLIC_SUPABASE_URL=https://uyrlqjpqidyilpcpdiqb.supabase.co
```
Comment (adityapawar1) [inline .env:1]: You should add this to the gitignore because we don't want the sensitive data in the repo! This is why we don't directly put these values in our code as variables.

File: app/PostComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  const [postData, setPostData] = useState<Post[]>([]);
+
+  async function fetchPostData() {
+    // TODO: Fetch data using the supabase client
+    const response = await supabase.from('posts').select('*');
+    const { data, error } = response;
+    if (error) {
+      throw error;
+    }
+    const formattedData: Post[] = data.map(post => ({
```
Comment (adityapawar1) [inline app/PostComponent.tsx:30]: I think you might have done this to get around some typescript error? Instead you should be able to omit this map and directly do `setPostData(data as Post[])` to cast the type.

File: app/PostComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+  }, []); // Empty dependency array means this effect runs only once
+
+  // Render the fetched data
+  return (
+    <div>
+      {postData == null ? (
+        <p>Loading...</p>
+      ) : (
+        postData.map(post => (
+          <main key={post.date} className={styles.main}>
```
Comment (adityapawar1) [inline app/PostComponent.tsx:55]: You should make this a Component! You can make a separate file with this code, add props (which will be similar to the post interface, and render it here. It'll make the code more readable.

File: app/page.tsx [hunk 10]
```diff
@@ -1,49 +1,14 @@
...
-          This organization is doing amazing work tackling the complex root
-          causes of the issue.
-        </p>
-
-        <ProfileIcon size={24} />
-        <p>vppraggie</p>
-        <p>September 21</p>
-        <p>Thanks for sharing!</p>
-      </div>
+      <PostComponent />
```
Comment (adityapawar1) [inline app/page.tsx:10]: A single post should be a component, with props similar to the Post interface. This is the entire post screen, so you can rename this to `<PostScreen>`

File: app/PostComponent.tsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+              <div className={styles.likesShareFooter}>
+                <HeartIcon size={24} />
+                {post.likes}
+                <ShareIcon size={24} />
+              </div>
+              <hr className={styles.separator} />
+              <div className={styles.commentContainer}>
+                <div className={styles.commentBox}>
+                  <div className={styles.userDateHeader}>
+                    <ProfileIcon size={24} />
```
Comment (adityapawar1) [inline app/PostComponent.tsx:77]: Based on the image you put in the PR, the styling looks a bit off. Make sure the styling looks as close to the figma as possible.

File: app/PostComponent.tsx [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+                  <p className={styles.commentContent}>
+                    This organization is doing amazing work tackling the complex
+                    root causes of the issue.
+                  </p>
+                </div>
+              </div>
+
+              <div className="comment-container">
+                <div className="comment-box">
```
Comment (adityapawar1) [inline app/PostComponent.tsx:90]: You should be using `className={styles.commentContainer}` etc. here instead of putting a class name. This will avoid classes in other files messing with the styles here.

**Review by adityapawar1 — CHANGES_REQUESTED** [review]
Small refactor and then it should be good to go!
